cpp/src/group_public_key_context.cpp

```diff
@@ -0,0 +1,109 @@
+/******************************************************************************
+ *
+ * Copyright 2018 Xaptum, Inc.
+ * 
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ * 
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+
+#include <xtt-cpp/group_public_key_context.hpp>
+
+#include <xtt/crypto_wrapper.h>
+
+#include <stdexcept>
+#include <vector>
+#include <algorithm>
+
+using namespace xtt;
+
+#include "internal/text_to_binary.hpp"
+
+const xtt_daa_group_pub_key_lrsw group_public_key_lrsw_dummy = {{0}};
+
+group_public_key_context_lrsw::group_public_key_context_lrsw()
+{
+    xtt_return_code_type ctor_ret =
+        xtt_initialize_group_public_key_context_lrsw(&gpk_ctx_,
+                                                     nullptr,
+                                                     0,
+                                                     &group_public_key_lrsw_dummy);
+    if (XTT_RETURN_SUCCESS != ctor_ret) {
+        throw std::runtime_error("Error encountered in LRSW group public key context constructor");
```

Comment:
Looks like the intent is that this call can never fail, so an exception should never be thrown?

Reply:
Right. It would probably be more appropriate to make this an `assert`.